ChromiumExtension/content/js/core.js

```diff
@@ -0,0 +1,480 @@
+                    key: 'Version',
+                    value: 'system/status'
+                },
+                {
+                    key: 'Status',
+                    value: 'system/status'
+                },
+                {
+                    key: 'QualityProfiles',
+                    value: 'qualityProfile'
+                }
+            ]
+        },
+        {
+            id: 'lidarr',
+            endpoints: [
+                {
+                    key: 'Version',
+                    value: 'v1/system/status'
+                },
+                {
+                    key: 'Status',
+                    value: 'v1/system/status'
+                },
+                {
+                    key: 'QualityProfiles',
+                    value: 'v1/qualityprofile'
+                }
+            ]
+        }
+    ],
+    versionConfig = [
+        {
+            id: 'sonarr',
+            configs: [
+                {
+                    match: /^2/,
+                    searchPath: '/addseries/',
+                    searchInputSelector: '.add-series-search .x-series-search'
+                },
+                {
+                    match: /^3/,
+                    searchPath: '/add/new/',
+                    searchInputSelector: 'input[name="seriesLookup"]'
+                }
+            ]
+        },
+        {
+            id: 'radarr',
+            configs: [
+                {
+                    match: /^0/,
+                    searchPath: '/addmovies/',
+                    searchInputSelector: '.add-movies-search .x-movies-search'
+                },
+                {
+                    match: /^3/,
+                    searchPath: '/add/new/',
+                    searchInputSelector: 'input[name="movieLookup"]'
+                }
+            ]
+        },
+        {
+            id: 'lidarr',
+            configs: [
+                {
+                    match: /^0/,
+                    searchPath: '/add/search/',
+                    searchInputSelector: 'input[name="searchBox"]'
+                }
+            ]
+        }
+    ];
+
+/**
+ * logs to console if the debug flag is set
+ * @param {any[]} content 
+ */
+var log = function(content) {
+    getSettings(function(settings) {
+        if (!settings.debug) {
+            return;
+        }
+    
+        console.log(content);
+    });
+}
+
+/*
+* Log old and new values when an item in a storage area is changed
+*/
+var logStorageChange = function(changes, area) {
+    let changedItems = Object.keys(changes);
+  
+    for (let item of changedItems) {
+        log(`Change in storage area: ${area} to item ${item}`);
+        log(['Old value: ', changes[item].oldValue]);
+        log(['New value: ', changes[item].newValue]);
+    }
+}
+
+chrome.storage.onChanged.addListener(logStorageChange);
+
+/**
+ * Retrieves settings from local storage
+ * @param {function} callback - function to call on completion
+ */
+var getSettings = function(callback) {
```

Comment:
Mostly a style question, but in general I prefer async/await over callbacks. I find it easier to read because it has fewer indentation levels.

And just an FYI, it is possible to mix and match the two styles, so even if `chrome.storage.sync.get` is callback-based, we could make `getSettings` Promise-based:
```
var getSettings = function() {
  return new Promise((resolve, reject) => {
    chrome.storage.sync.get({ 'sonarrRadarrLidarrAutosearchSettings': defaultSettings }, function(data) {
      // do things with data
      resolve(data)
  }
}

var settings = await getSettings();
```

With that said, I can grok callbacks as well. So I'm OK with continuing to use callbacks, just wanted to raise the topic :)

(And a follow-up question: if I contribute code in the future, would you prefer it to be callback-based or can I use promises?)

Reply:
I agree with you regarding callbacks vs promises and was thinking of doing that, however there are some considerations. I wanted to keep both Firefox and Chrome extensions in line, mainly because there's a polyfill lib that can be used on a Firefox add on and I thought this might be best going forward: https://extensionworkshop.com/documentation/develop/browser-compatibility/